joern-cli/frontends/php2cpg/src/test/scala/io/joern/php2cpg/querying/FileTests.scala

```diff
@@ -0,0 +1,54 @@
+package io.joern.php2cpg.querying
+
+import io.joern.php2cpg.testfixtures.PhpCode2CpgFixture
+import io.joern.php2cpg.utils.PathFilter
+import io.joern.x2cpg.testfixtures.TestCpg
+import io.shiftleft.semanticcpg.language._
+import io.joern.php2cpg.Config
+
+class FileTests extends PhpCode2CpgFixture {
+  val testFiles = List(
+    "a.php",
+    "src/b.php",
+    "src/sub/c.php",
+    "tests/d.php",
+    "tests/sub/e.php",
+    "vendor/autoload.php",
+    "vendor/dep/f.php",
+    "misc/g.php",
+    ".git/h.php",
+    "sub/tests/i.php",
+    "nottests/j.php"
+  )
+
+  private def getCpg(): TestCpg = {
+    val cpg = code("", fileName = testFiles.head)
+    testFiles.tail.foreach(path => cpg.moreCode("", fileName = PathFilter.standardisePath(path)))
+    cpg
+  }
+
+  "test, vendor, and .git directories should be filtered out by default" in {
+    val cpg = getCpg()
+
+    cpg.file.name.toSet shouldBe Set("a.php", "src/b.php", "src/sub/c.php", "misc/g.php", "nottests/j.php", "<unknown>")
+      .map(PathFilter.standardisePath)
+  }
+
+  "custom excludes should exclude the correct directories" in {
+    val config = Config(excludeOverrides = Some(List("dep", "src/sub/", "doesntexist")))
+    val cpg    = getCpg().config(config)
+
+    cpg.file.name.toSet shouldBe Set(
+      "a.php",
+      "src/b.php",
+      "tests/d.php",
+      "tests/sub/e.php",
+      "vendor/autoload.php",
+      "misc/g.php",
+      ".git/h.php",
+      "sub/tests/i.php",
+      "nottests/j.php",
+      "<unknown>"
+    ).map(PathFilter.standardisePath)
+  }
+}
```

Comment:
with other frontends we've seen bugs with filtering for untested combinations of absolute and relative paths for a) the input path and b) the excluded path. Can you add some tests for here to ensure that works as intended in all cases?